lib/rtcpeerconnection/edge.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+var EventTarget = require('../util/eventtarget');
+var inherits = require('util').inherits;
+var util = require('../util');
+
+var match = navigator.userAgent.match(/Edge\/(\d+).(\d+)$/);
+var PeerConnection = require('rtcpeerconnection-shim')(window, match[2]);
+
+function EdgeRTCPeerConnection(configuration) {
+  if (!(this instanceof EdgeRTCPeerConnection)) {
+    return new EdgeRTCPeerConnection(configuration);
+  }
+
+  EventTarget.call(this);
+
+  var peerConnection = new PeerConnection(configuration);
+
+  Object.defineProperties(this, {
+    _peerConnection: {
+      value: peerConnection
+    },
+    iceConnectionState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceConnectionState;
+      }
+    },
+    iceGatheringState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceGatheringState;
+      }
+    },
+    signalingState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.signalingState;
+      }
+    }
```

Comment:
Can you not depend on `util.proxyProperties` to handle these? Specifically `iceConnectionState`, `iceGatheringState`, and `signalingState`.

Reply:
Those properties are not defined on Prototype but on Object instance. I seen some commits in upstream defining those properties on Prototype. I will leave a note pointing to the commit.